### DISCUSSION TITLE
Security with activePivot

### DISCUSSION THREAD

**Discussion by sjmg** [atoti/atoti#700] [Q&A]
Hello,

We are using in the same Active UI 5, ActivePivot and Atoti+.  
Nowadays we have implemented the security access with ActivePivot and we want to know if its possible to get the user information from ActivePivot session to Atoti

Thanks so much

Regards

**Comment by tibdex** [comment]
Hi, I'm not sure to understand what you mean by "get the user information from ActivePivot session to Atoti". Can you elaborate a bit more?

**Comment by sjmg** [comment]
Hello,

We have an ActiveUI5 with an activePivot and Atoti cubes. 
We are check the security in activePivot and we want to share the information of this login between ActivePivot and Atoti (username, roles)

It's possible to share this information?

Thanks

**Reply by tibdex** [reply]
To do that you'll need to manually keep in sync the roles and restrictions config between ActivePivot and atoti. There is no automated way to do this for now.

**Reply by mwebborn-av** [reply]
If the content server is provided by the ActivePivot cube, could we in theory use the JWT returned from there to login to the atoti cube?

The root issue here is that we cannot recreate the security implementation from the Java API in atoti as it stands, so we want to work around this by authenticating through the existing AP cubes. For reference, the two blockers for implementing the same security will probably be solved by these issues: https://github.com/activeviam/atoti/issues/4057 and 
https://github.com/activeviam/atoti/issues/3334

**Reply by Skyscimitar** [reply]
If you configure the atoti session to use the same key files using [`tt.JwtConfig`](https://docs.atoti.io/latest/lib/atoti/atoti.config.jwt_config.html#atoti.JwtConfig)  so that both servers use the same key to sign their tokens, then you will be able to authenticate users with the JWT returned by the content server by passing the JWT in an Authorization header. However you will still have to configure roles and restrictions for the users in the atoti session, there is currently no way of keeping these in sync automatically.
As you pointed out though, the PRs related to activeviam/atoti#3334 should resolve the synchronization problems between all the servers.
